discovery_engine/test/integration/search_test.dart

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io' show Directory;
+
+import 'package:test/test.dart';
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show
+        ActiveSearchClosedFailed,
+        ActiveSearchClosedSucceeded,
+        ActiveSearchRequestFailed,
+        ActiveSearchRequestSucceeded,
+        ActiveSearchTermRequestFailed,
+        ActiveSearchTermRequestSucceeded,
+        DiscoveryEngine,
+        NextActiveSearchBatchRequestFailed,
+        NextActiveSearchBatchRequestSucceeded,
+        RestoreActiveSearchSucceeded,
+        SearchBy,
+        SearchFailureReason;
+
+import '../logging.dart' show setupLogging;
+import 'utils/helpers.dart'
+    show TestEngineData, expectEvent, initEngine, setupTestEngineData;
+import 'utils/local_newsapi_server.dart' show LocalNewsApiServer;
+
+void main() {
+  setupLogging();
+
+  const searchTerm = 'birds eat seeds';
+  const searchTerm2 = 'birds also eat insects';
+
+  group('News Search', () {
+    late LocalNewsApiServer server;
+    late TestEngineData data;
+    late DiscoveryEngine engine;
+
+    setUp(() async {
+      data = await setupTestEngineData();
+      server = await LocalNewsApiServer.start();
+      engine = await initEngine(data, server.port);
+    });
+
+    tearDown(() async {
+      await engine.dispose();
+      await server.close();
+      await Directory(data.applicationDirectoryPath).delete(recursive: true);
+    });
+
+    test('a new search can be opened', () async {
+      final result = expectEvent<ActiveSearchRequestSucceeded>(
```

Comment:
:art: perhaps worth checking the latest re naming of "active search" - I seem to recall people in favour of dropping the "active". but this would prob be a separate pr

Reply:
This is why the file is  `search_test.dart`. Through `ActiveSearchRequestSucceeded` is auto generated by freezed and part of the team blue public API. We probably won't rename it until switching to the message passing API. But even if we rename it earlier it has nothing to do with this PR.